src/main/java/io/jenkins/plugins/artifact_manager_s3/S3BlobStore.java

```diff
@@ -60,37 +62,45 @@
  * Extension that customizes JCloudsBlobStore for AWS S3. Credentials are fetched from the environment, env vars, aws
  * profiles,...
  */
-@Extension
 @Restricted(NoExternalUse.class)
-public class S3BlobStore extends JCloudsApiExtensionPoint {
+public class S3BlobStore extends BlobStoreProvider {
 
     private static final Logger LOGGER = Logger.getLogger(S3BlobStore.class.getName());
 
     private static final long serialVersionUID = -8864075675579867370L;
 
+    // For now, these are taken from the environment, rather than being configured.
+    @SuppressWarnings("FieldMayBeFinal")
+    private static String BLOB_CONTAINER = System.getenv("S3_BUCKET");
```

Comment:
why make them S3 specific when affect any future providers too?

Reply:
Just to be on the conservative side—we do not know yet if configuration of container and/or prefix will be desirable or necessary for other cloud providers. This API allows the provider to decide whether to hard-code values; compute them from some more general settings; get them from environment variables (system properties, …); or let the user select them in `/configure`, presumably with form validation and completion and so on, possibly alongside credentials.